deploy/charts/cert-manager/values.yaml

```diff
@@ -148,6 +148,18 @@ prometheus:
     interval: 60s
     scrapeTimeout: 30s
     labels: {}
+  prometheusrule:
+    enabled: false
+    labels: {}
+    rules:
+    - alert: CertManagerAbsent
```

Comment:
It seems weird to me that we are basically embedding the PrometheusRule spec here - why would a user choose to embed their AlertManager alerts into our values.yaml, when it does not have a schema, as well as less control over the resulting resource that is created? To me at least, it seems like it'd make more sense to have your own standalone chart/set of YAML for this. It doesn't seem like this provides a meaningful/useful abstraction to users...

Reply:
As an user I expect the cert-manager helm chart to provide me with basic alerts grouped as application (as a feature flag). This is a good start point to create a basic set of alerts.

It can also be used to find alerts that make sense for cert-manager that I can use in my prometheus operator setup.

Reply:
Sure, but I don't think exposing the entirety of the `PrometheusRule` spec is desirable, as this basically makes the cert-manager Helm chart a 'deployment tool' for PrometheusRule resources. I agree that it's great if we can provide some out of the box configuration to guide people & to get them started though.

Instead of exposing all this in `values.yaml`, can we instead define this as a resource in the `templates/` directory and gate it behind a simple boolean `prometheus.alerts.enabled: true/false`? (with a default of `false` as a lot of users won't have PrometheusRule as a CRD installed?)

I think that'd make this a lot more palatable to accept, and if in future there is demand for users being able to configure their own custom alerts *via our Helm chart*, we are not 'boxing ourselves in' and it'd be possible to still add `prometheus.alerts.customRules` or something?

Reply:
Usually, what is done is that there are default rules (with `prometheusRule.enabled: false` by default) with specific selectors + templating in order NOT to be triggered by another cert-manager, and the whole `PrometheusRule` value is given. Of course, it has weak validation from the Chart point of view, but anyway Prometheus validator webhook is able to do the validation.

See bitnami for example: https://github.com/bitnami/charts/blob/master/bitnami/redis/values.yaml#L1222

It would be very useful to have such default alerts to have alerting out of the box. Today, everybody have to re-implement them.

Reply:
I understand your remarks @munnerz. I'll update the PR to reflect your idea.